api/schema/resolvers/PaymentResolver.js

```diff
@@ -77,6 +78,32 @@ module.exports = {
         deletePaymentById: (root, { id }, { models }) => {
             return models.Payment.destroy({ where: { id } })
         },
+        generateBitcoinInvoiceFromPayment: async (root, { paymentId }, { models }) => {
+            
+            const Payment = await models.Payment.findByPk(paymentId)
+            const { amount, client_id, date_paid, external_uuid, external_uuid_type } = Payment.dataValues
+
+            const client = await models.Client.findByPk(client_id)
+            
+            const isClientCurrencyBtc = client.dataValues.currency === 'BTC' || client.dataValues.currency === 'SATS'
+            if (!isClientCurrencyBtc) throw 'Client\'s currency is not Bitcoin'
+
+            const isInvoiceExpired = external_uuid && external_uuid_type === 'bitcoin' && await checkInvoiceExpired(external_uuid)
+            if (external_uuid && date_paid && !isInvoiceExpired) throw 'An active invoice already exists'
+
+            const amountInSats = client.dataValues.currency === 'BTC' ? amount * 1000000 : amount / 100;
+            
+            const newInvoice = await createBitcoinInvoice(amountInSats)
+            return models.Payment.update({
+                external_uuid: newInvoice.id,
+                external_uuid_type: 'bitcoin'
+            }, 
+            { where: 
+                {
+                    id: paymentId
+                } 
+            })
```

Comment:
Let's also refactor all of this logic into an API module so we don't tons of logic in our GraphQL resolvers

As a separation of concerns I would recommend resolvers be reserved for simple logic that is needed for tweaking incoming arguments before sending it to a function in `/api/modules/...` or perhaps tweaking the outgoing response but most of the core logic, especially making requests to other files like the `btcPayServer` handler

This way our resolvers depend on API modules and nothing else and the modules can have further dependencies